erc20_example/src/erc20_actor.rs

```diff
@@ -0,0 +1,269 @@
+
+fil_actors_runtime::wasm_trampoline!(Actor);
+
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+    Name = frc42_dispatch::method_hash!("Name"),
+    Symbol = frc42_dispatch::method_hash!("Symbol"),
+    Decimals = frc42_dispatch::method_hash!("Decimals"),
+    TotalSupply = frc42_dispatch::method_hash!("TotalSupply"),
+    BalanceOf = frc42_dispatch::method_hash!("BalanceOf"),
+    Transfer = frc42_dispatch::method_hash!("Transfer"),
+    Allowance = frc42_dispatch::method_hash!("Allowance"),
+    Approve = frc42_dispatch::method_hash!("Approve"),
+    TransferFrom = frc42_dispatch::method_hash!("TransferFrom"),
+    IncreaseAllowance = frc42_dispatch::method_hash!("IncreaseAllowance"),
+    DecreaseAllowance = frc42_dispatch::method_hash!("DecreaseAllowance"),
+    Mint = frc42_dispatch::method_hash!("Mint"),
+    Burn = frc42_dispatch::method_hash!("Burn"),
+}
+
+pub trait MyToken {
+    fn constructor(rt: &impl Runtime) -> Result<(), ActorError>;
+
+    fn name(rt: &impl Runtime) -> Result<String, ActorError>;
+
+    fn symbol(rt: &impl Runtime) -> Result<String, ActorError>;
+
+    fn decimals(rt: &impl Runtime) -> Result<u8, ActorError>;
+
+    fn total_supply(rt: &impl Runtime) -> Result<BigUint, ActorError>;
+
+    fn balance_of(rt: &impl Runtime, account: Address) -> Result<BigUint, ActorError>;
+
+    fn transfer(rt: &impl Runtime, transfer_params: Params) -> Result<(), ActorError>;
+
+    fn allowance(
+        rt: &impl Runtime,
+        allowance_params: AllowanceParams,
+    ) -> Result<BigUint, ActorError>;
+
+    fn approve(rt: &impl Runtime, approve_params: Params) -> Result<(), ActorError>;
+
+    fn transfer_from(
+        rt: &impl Runtime,
+        transfer_from_params: TransferFromParams,
+    ) -> Result<(), ActorError>;
+
+    fn increase_allowance(
+        rt: &impl Runtime,
+        increase_allowance_params: Params,
+    ) -> Result<(), ActorError>;
+
+    fn decrease_allowance(
+        rt: &impl Runtime,
+        decrease_allowance_params: Params,
+    ) -> Result<(), ActorError>;
+
+    //added to the actor trait for testing purpose
+    fn mint(rt: &impl Runtime, mint_params: Params) -> Result<(), ActorError>;
+
+    //added to the actor trait for testing purpose
+    fn burn(rt: &impl Runtime, burn_params: Params) -> Result<(), ActorError>;
+}
+
+pub struct Actor;
+
+impl MyToken for Actor {
+    fn constructor(rt: &impl Runtime) -> Result<(), ActorError> {
+        rt.validate_immediate_caller_is(std::iter::once(&INIT_ACTOR_ADDR))?;
+
+        let params = ConstructParams {
+            name: String::from("MyToken"),
+            sybmol: String::from("MT"),
+        };
+
+        let st = State::new(rt.store(), params).map_err(|e| {
+            e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "Failed to create actor state.")
+        })?;
+
+        rt.create(&st)?;
+
+        let st: State = rt.state()?;
+
+        //10^18
+        let amount = 10
```

Comment:
With fvm_shared BigUint library you can create amount like this instead
```suggestion
        let amount = BigInt::from(10)
            .pow(u32::from(st.erc20_base.decimals()));
```

Reply:
Fixed [here](https://github.com/Ethernal-Tech/filecoin-actor-library/pull/9/commits/8f4a670cb13b535b176f0710e5c46cab9dab6f7e#diff-0f4d54b27cfb82e375b18c832727afb516d280df84b3536800b850e7e1647959L98).